data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/S3ObjectReference.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import java.util.Objects;
+
+public class S3ObjectReference {
```

Comment:
You might get a merge conflict on this file since I also added this into `main`. Please use the one I provided as it has `toString()` and is unit tested.